ethportal-api/src/types/cli.rs

```diff
@@ -709,6 +804,247 @@ mod test {
+            assert_eq!(
+                config.storage_capacity_config(),
+                StorageCapacityConfig::Combined {
+                    total_mb: 200,
+                    subnetworks: vec![Subnetwork::Beacon, Subnetwork::History, Subnetwork::State],
+                }
+            );
+        }
+
+        #[test]
+        #[should_panic(
+            expected = "--storage.total and --storage.beacon can't be set at the same time"
+        )]
+        fn with_total_and_beacon() {
+            TrinConfig::new_from(["trin", "--storage.total", "200", "--storage.beacon", "100"])
+                .unwrap();
+        }
+
+        #[test]
+        #[should_panic(
+            expected = "--storage.total and --storage.history can't be set at the same time"
+        )]
+        fn with_total_and_history() {
+            TrinConfig::new_from(["trin", "--storage.total", "200", "--storage.history", "100"])
+                .unwrap();
+        }
+
+        #[test]
+        #[should_panic(
+            expected = "--storage.total and --storage.state can't be set at the same time"
+        )]
+        fn with_total_and_state() {
+            TrinConfig::new_from(["trin", "--storage.total", "200", "--storage.state", "100"])
+                .unwrap();
+        }
+
+        #[test]
+        fn with_history() {
+            let config = TrinConfig::new_from(["trin", "--storage.history", "200"]).unwrap();
+            assert_eq!(
+                config.storage_capacity_config(),
+                StorageCapacityConfig::Specific {
+                    beacon_mb: None,
+                    history_mb: Some(200),
+                    state_mb: None,
+                }
+            );
+        }
+
+        #[test]
+        fn with_history_and_state() {
+            let config = TrinConfig::new_from([
+                "trin",
+                "--storage.history",
+                "200",
+                "--storage.state",
+                "300",
+                "--portal-subnetworks",
+                "history,state",
+            ])
+            .unwrap();
+            assert_eq!(
+                config.storage_capacity_config(),
+                StorageCapacityConfig::Specific {
+                    beacon_mb: None,
+                    history_mb: Some(200),
+                    state_mb: Some(300),
+                }
+            );
+        }
+
+        #[test]
+        fn with_history_and_state_and_beacon() {
+            let config = TrinConfig::new_from([
+                "trin",
+                "--storage.history",
+                "200",
+                "--storage.state",
+                "300",
+                "--storage.beacon",
+                "0",
+                "--portal-subnetworks",
+                "history,state,beacon",
+            ])
+            .unwrap();
+            assert_eq!(
+                config.storage_capacity_config(),
+                StorageCapacityConfig::Specific {
+                    beacon_mb: Some(0),
```

Comment:
I don't think there are any actual bugs here, but as a practice the test for setting a specific value should probably pick one that isn't the value you would arrive at by default.